src/protocols/secure_channel/CASESession.cpp

```diff
@@ -1662,6 +1695,22 @@ CHIP_ERROR CASESession::OnMessageReceived(ExchangeContext * ec, const PayloadHea
     Protocols::SecureChannel::MsgType msgType = static_cast<Protocols::SecureChannel::MsgType>(payloadHeader.GetMessageType());
     SuccessOrExit(err);
 
+#if CONFIG_IM_BUILD_FOR_UNIT_TEST
```

Comment:
`CONFIG_IM_BUILD_FOR_UNIT_TEST` was recommended to me to compile test specific function to intercept the handshake. I was under the impression that `TestCASESession.cpp` was a host test only, looks like it might also be a device side test (or it is at least compiled for device targets). Not sure what is the best way have this method on available tests, so would welcome suggestions.

Reply:
Conditioning things on CONFIG_IM_BUILD_FOR_UNIT_TEST that would never be done in production makes a lot of sense.